lib/AST/ASTImporter.cpp

```diff
@@ -4524,14 +4523,8 @@ Decl *ASTNodeImporter::VisitClassTemplateDecl(ClassTemplateDecl *D) {
 
   ToTemplated->setDescribedClassTemplate(D2);
 
-  if (PrevDecl) {
-    PrevDecl = PrevDecl->getMostRecentDecl();
-    assert(
-      (!ToTemplated->getPreviousDecl() ||
-      ToTemplated->getPreviousDecl()->getDescribedClassTemplate() == PrevDecl)
-      && "Experimental check failed");
-    D2->setPreviousDecl(PrevDecl);
-  }
+  if (ToTemplated->getPreviousDecl())
+    D2->setPreviousDecl(ToTemplated->getPreviousDecl()->getDescribedClassTemplate());
```

Comment:
Is it always the case that we have a describedClassTemplate in the previousDecl?
Perhaps we should add an assert before this line:
```
assert(ToTemplated->getPreviousDecl()->getDescribedClassTemplate());
```

Or  in that case the previous decl will be simply a nullptr and the chain will not be built. Is it okay?

Reply:
A decl-chain of a "templated" record should contain only "templated" records (at least I think). (Or are the specializations other case?) Probably we can "walk" with getPreviousDecl until there is a getDescribedClassTemplate.